python/src/aiconfig/editor/client/src/reducers/aiconfigReducer.ts

```diff
@@ -167,18 +167,27 @@ export default function aiconfigReducer(
       }));
     }
     case "UPDATE_PROMPT_MODEL": {
-      return reduceReplacePrompt(dirtyState, action.id, (prompt) => ({
-        ...prompt,
-        metadata: {
-          ...prompt.metadata,
-          model: action.modelName
-            ? {
-                name: action.modelName,
-                // TODO: Consolidate settings based on schema union
-              }
-            : undefined,
-        },
-      }));
+      return reduceReplacePrompt(dirtyState, action.id, (prompt) => {
+        // TODO: Consolidate settings based on schema union, server-side
+        // For now, just keep all the settings to match the server-side implementation
+        let modelSettings;
+        const promptModel = prompt.metadata?.model;
+        if (promptModel && typeof promptModel !== "string") {
+          modelSettings = promptModel.settings;
+        }
+        return {
+          ...prompt,
+          metadata: {
+            ...prompt.metadata,
+            model: action.modelName
+              ? {
+                  name: action.modelName,
+                  settings: modelSettings,
```

Comment:
we can just do `const` on L176 and this wil be undefined if never set right?

Reply:
I did let with the if instead of using a const with huge ternary. I feel pretty strongly about not initializing as undefined since the above is pretty standard in js and explicitly defining it to undefined is really weird (you would in fact be defining it ... as undefined lol)